pymongo/database.py

```diff
@@ -292,6 +292,26 @@ def get_collection(
             read_concern,
         )
 
+    def _get_encrypted_fields(self, kwargs, coll_name, ask_db):
+        encrypted_fields = kwargs.get("encryptedFields")
+        if encrypted_fields:
+            return encrypted_fields.copy()
```

Comment:
These should be deep copies otherwise we'll still be mutating the underlying dictionaries.

Reply:
Done and added fixed the test.